src/tests/functional/shared_test_classes/src/subgraph/multiple_LSTMCell.cpp

```diff
@@ -383,94 +443,82 @@ void MultipleLSTMCellTest::CreatePureTensorIteratorModel() {
+            std::memcpy(tensor.data(), cell_memory_init.data(), cell_memory_init.size() * sizeof(float));
+        } else if (name.find("hidden_state_1") != std::string::npos ||
+                   name.find("hidden_state_2") != std::string::npos) {
+            std::memcpy(tensor.data(), hidden_memory_init.data(), hidden_memory_init.size() * sizeof(float));
         } else {
             GTEST_FAIL() << "unknown memory state";
         }
+        state.set_state(tensor);
     }
 }
 
-void MultipleLSTMCellTest::ApplyLowLatency() {
+void MultipleLSTMCellTest::apply_low_latency() {
     // Calculate values after LowLatency transformation
-    CreatePureTensorIteratorModel();
-    if (transformation == ngraph::helpers::MemoryTransformation::LOW_LATENCY_V2) {
+    create_pure_tensor_iterator_model();
+    function = functionRefs;
+    if (transformation == ov::test::utils::MemoryTransformation::LOW_LATENCY_V2) {
         function->validate_nodes_and_infer_types();
         // Apply LowLatency (insert Assigns/ReadValues) and UnrollTensorIterator
 
-        pass::Manager manager;
+        ov::pass::Manager manager;
         manager.register_pass<pass::LowLatency2>();
         manager.run_passes(function);
-        bool ti_found = helpers::is_tensor_iterator_exist(function);
+        bool ti_found = ngraph::helpers::is_tensor_iterator_exist(function);
         EXPECT_EQ(ti_found, false);
-        LoadNetwork();
-    } else if (transformation == ngraph::helpers::MemoryTransformation::LOW_LATENCY_V2_REGULAR_API) {
-        cnnNetwork = InferenceEngine::CNNNetwork{function};
-        InferenceEngine::lowLatency2(cnnNetwork);
-
-        bool ti_found = helpers::is_tensor_iterator_exist(cnnNetwork.getFunction());
-        EXPECT_EQ(ti_found, false);
-
-        ConfigureNetwork();
-        executableNetwork = core->LoadNetwork(cnnNetwork, targetDevice, configuration);
-        inferRequest = executableNetwork.CreateInferRequest();
+        compile_model();
     }
 }
 
-void MultipleLSTMCellTest::LoadNetwork() {
-    LayerTestsUtils::LayerTestsCommon::LoadNetwork();
-    inferRequest = executableNetwork.CreateInferRequest();
+void MultipleLSTMCellTest::compile_model() {
+    ov::test::SubgraphBaseStaticTest::compile_model();
+    inferRequest = compiledModel.create_infer_request();
 }
 
-void MultipleLSTMCellTest::Run() {
+void MultipleLSTMCellTest::infer() {
+    for (const auto& input : inputs) {
+        inferRequest.set_tensor(input.first, input.second);
+    }
+    inferRequest.infer();
+}
+
+void MultipleLSTMCellTest::run() {
     SKIP_IF_CURRENT_TEST_IS_DISABLED()
-    if (transformation != ngraph::helpers::MemoryTransformation::NONE) {
-        ApplyLowLatency();
+    if (transformation != ov::test::utils::MemoryTransformation::NONE) {
+        apply_low_latency();
     } else {
-        LoadNetwork();
+        compile_model();
     }
 
-    InitMemory();
-    GenerateInputs();
-    Infer();
+    init_memory();
+    generate_inputs(input_shapes);
```

Comment:
no `infer` call here

Reply:
`infer `has been called at:
https://github.com/openvinotoolkit/openvino/blob/61c5a80fa63ed3dd7eb353c62ceee908ef9d8d69/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp#L458

Not need call twice `infer`, because LSTM is statefull model, the second `infer `result will be different with the first `infer `calling.

Reply:
Looks like there was some intention to call inference twice because `switch_to_friendly_model()` and `create_pure_tensor_iterator_model` changes `function` which will be inferred in `validate()`

@itikhono please, help with review

Reply:
`switch_to_friendly_model()` and `create_pure_tensor_iterator_model` should return `refFunction `and will be inferred by `TEMPLATE ` plugin.
`validate()` will call `infer()` for setup model and refFunction separately:

Infer for target device:
https://github.com/openvinotoolkit/openvino/blob/d1f71ee233f22e4af1dac32105f25161808b1c63/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp#L452-L463

Infer for refFunction:
https://github.com/openvinotoolkit/openvino/blob/d1f71ee233f22e4af1dac32105f25161808b1c63/src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp#L424-L443

Reply:
Got it!